python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -644,49 +647,58 @@ class RandomForestClassifier(Base):
-                    This is optional and required only while performing the
-                    predict operation on the GPU.
-                    It is applied if output_class == True, else it is ignored
-        num_classes : integer
+        predict_model : String (default = 'GPU')
+            'GPU' to predict using the GPU, 'CPU' otherwise. The GPU can only
+            be used if the model was trained on float32 data and `X` is float32
+            or convert_dtype is set to True.
+        output_class: boolean (default = True)
+            This is optional and required only while performing the
+            predict operation on the GPU.
+            If true, return a 1 or 0 depending on whether the raw
+            prediction exceeds the threshold. If False, just return
+            the raw prediction.
+        algo : string (default = 'BATCH_TREE_REORG')
+            This is optional and required only while performing the
+            predict operation on the GPU.
+            'NAIVE' - simple inference using shared memory
+            'TREE_REORG' - similar to naive but trees rearranged to be more
+                           coalescing-friendly
+            'BATCH_TREE_REORG' - similar to TREE_REORG but predicting
+                                 multiple rows per thread block
+        threshold : float (default = 0.5)
+            Threshold used for classification. Optional and required only
+            while performing the predict operation on the GPU.
+            It is applied if output_class == True, else it is ignored
+        num_classes : int (default = 2)
                       number of different classes present in the dataset
+        convert_dtype : bool, optional (default = False)
+            When set to True, the predict method will, when necessary, convert
+            the input to the data type which was used to train the model. This
+            will increase memory used for the method.
 
         Returns
         ----------
         y : NumPy
            Dense vector (int) of shape (n_samples, 1)
         """
-        if self.dtype == np.float64:
-            raise TypeError("GPU predict model only accepts float32 dtype"
-                            " as input, convert the data to float32 or "
-                            "use the CPU predict with `predict_model='CPU'`.")
 
-        elif predict_model == "CPU" or predict_model is None:
-            preds = self._predict_model_on_cpu(X)
+        if predict_model == "CPU":
+            preds = self._predict_model_on_cpu(X, convert_dtype)
+
+        elif self.dtype == np.float64 and convert_dtype is False:
+            warnings.watn("Using CPU based predict. \
+                          GPU predict model only accepts float32 and \
+                          model was trained with float64. TO use GPU based \
+                          predict, convert `X` to float32 or set the \
+                          parameter convert_dtype to True to have the \
```

Comment:
Is this right that convert_dtype will work? I think the issue is with `self.dtype` not X.type. If self.dtype is float64, won't fil complain.

Reply:
We should add a test that trains in float64 and predicts in float32 to confirm

Reply:
it'll work as in it'll try to convert the dtype if its not the same as self.dtype, which is the standard way as all the other algos. It was missing totally any usage of that. @Salonijain27 will be adding a test

Reply:
Should we have an automatic check in predict (convert_dtype = True by default) that converts the data to the dtype used for fit? This way we will be able to use different dtype for prediction and fit like sklearn does.

Reply:
I think what you mean is whether the default for convert_dtype should be True, otherwise what you describe is alrady the case, just that the user has to set it to True expliclty